tests/suite/reconfig_test.go

```diff
@@ -0,0 +1,638 @@
+		}
+
+		Expect(resourceManager.Apply([]client.Object{&reconfigNamespace})).To(Succeed())
+		Expect(resourceManager.ApplyFromFiles(
+			[]string{
+				"reconfig/cafe-secret.yaml",
+				"reconfig/reference-grant.yaml",
+			},
+			reconfigNamespace.Name)).To(Succeed())
+
+		Expect(createUniqueResources(resourceCount, "manifests/reconfig/cafe.yaml")).To(Succeed())
+
+		Expect(createUniqueResources(resourceCount, "manifests/reconfig/cafe-routes.yaml")).To(Succeed())
+
+		for i := 1; i <= resourceCount; i++ {
+			ns := core.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "namespace" + strconv.Itoa(i),
+				},
+			}
+			Expect(resourceManager.WaitForPodsToBeReady(ctx, ns.Name)).To(Succeed())
+		}
+
+		Expect(resourceManager.ApplyFromFiles([]string{"reconfig/gateway.yaml"}, reconfigNamespace.Name)).To(Succeed())
+
+		return nil
+	}
+
+	createResourcesRoutesLast := func(resourceCount int) error {
+		ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.CreateTimeout)
+		defer cancel()
+
+		for i := 1; i <= resourceCount; i++ {
+			ns := core.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "namespace" + strconv.Itoa(i),
+				},
+			}
+			Expect(k8sClient.Create(ctx, &ns)).To(Succeed())
+		}
+
+		Expect(createUniqueResources(resourceCount, "manifests/reconfig/cafe.yaml")).To(Succeed())
+
+		for i := 1; i <= resourceCount; i++ {
+			ns := core.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "namespace" + strconv.Itoa(i),
+				},
+			}
+			Expect(resourceManager.WaitForPodsToBeReady(ctx, ns.Name)).To(Succeed())
+		}
+
+		Expect(resourceManager.Apply([]client.Object{&reconfigNamespace})).To(Succeed())
+		Expect(resourceManager.ApplyFromFiles(
+			[]string{
+				"reconfig/cafe-secret.yaml",
+				"reconfig/reference-grant.yaml",
+				"reconfig/gateway.yaml",
+			},
+			reconfigNamespace.Name)).To(Succeed())
+
+		Expect(createUniqueResources(resourceCount, "manifests/reconfig/cafe-routes.yaml")).To(Succeed())
+
+		return nil
+	}
+
+	checkResourceCreation := func(resourceCount int) error {
+		ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.GetTimeout)
+		defer cancel()
+
+		var namespaces core.NamespaceList
+		if err := k8sClient.List(ctx, &namespaces); err != nil {
+			return fmt.Errorf("error getting namespaces: %w", err)
+		}
+		Expect(len(namespaces.Items)).To(BeNumerically(">=", resourceCount))
+
+		var routes v1.HTTPRouteList
+		if err := k8sClient.List(ctx, &routes); err != nil {
+			return fmt.Errorf("error getting HTTPRoutes: %w", err)
+		}
+		Expect(len(routes.Items)).To(BeNumerically("==", resourceCount*3))
+
+		var pods core.PodList
+		if err := k8sClient.List(ctx, &pods); err != nil {
+			return fmt.Errorf("error getting Pods: %w", err)
+		}
+		Expect(len(pods.Items)).To(BeNumerically(">=", resourceCount*2))
+
+		return nil
+	}
+
+	cleanupResources := func() error {
+		var err error
+
+		for i := 1; i <= maxResourceCount; i++ {
+			nsName := "namespace" + strconv.Itoa(i)
```

Comment:
it's significantly faster to delete the namespaces together in one command:

```
kubectl delete ns1 ns2 ns3 ns4 ns5 
```

For example, when deleting just 7 namespaces individually, it takes 39 seconds:
```
 time bash -c "kubectl delete ns ns1 && kubectl delete ns ns2 && kubectl delete ns ns3 && kubectl delete ns ns4 && kubectl delete ns ns5 && kubectl delete ns ns6 && kubectl delete ns ns7"                 10:00:09
namespace "ns1" deleted
namespace "ns2" deleted
namespace "ns3" deleted
namespace "ns4" deleted
namespace "ns5" deleted
namespace "ns6" deleted
namespace "ns7" deleted
bash -c   0.73s user 0.39s system 2% cpu 39.253 total
```

vs, 8 seconds:

```
time kubectl delete ns ns1 ns2 ns3 ns4 ns5 ns6 ns7                                                                                                                             10:00:23
namespace "ns1" deleted
namespace "ns2" deleted
namespace "ns3" deleted
namespace "ns4" deleted
namespace "ns5" deleted
namespace "ns6" deleted
namespace "ns7" deleted
kubectl delete ns ns1 ns2 ns3 ns4 ns5 ns6 ns7  0.08s user 0.06s system 2% cpu 5.568 total
```

Reply:
I'll look into this, but is this a case where I should use a bash command or `DeleteAllOf` function in the client package?

Reply:
The `DeleteAllOf` function would be ideal if you can get it to work. You may need to add a label to all of the namespaces you create so you can select on the label when deleting all.

Reply:
This should hopefully significantly reduce the time it takes to run this function.

Reply:
I've been exploring the `DeleteAllOf` however I get this error: 

`the server does not allow this method on the requested resource`

When I do some researching I think that `DeleteAllOf` could be using / has the same logic/reasoning as the "deletecollection" verb which I don't think namespaces has:

`kubectl api-resources --sort-by name -o wide`'s result shows:

`namespaces          ns     v1         false     Namespace       create,delete,get,list,patch,update,watch`

I can't seem to get it to work and the above I think is an explanation.

Reply:
I see. We may not be able to do this with the client. Let's just add an issue to research this further and link to the issue in the code. If we need to cut down on the total runtime of the NFR tests, this optimization will help.

Reply:
Should I add a TODO in the code?